2.0 Compatibility, subjects, and localization

DrDudelsack opened the pull request:
Minor tweaks in main.java for 2.0 compatibility, the addition of subjects being converted as vassals (later some types of subjects will be converted as tributaries), minor culture mapping changes, and preliminary localization for the cultures, religions, and bookmark

DrDudelsack committed:
2.0 compatibility, subjects, localization

Basic tweaks to allow 2.0 compatibility, updated culture map, subjects are now converted as vassals, and localization for the bookmarks

DrDudelsack committed:
Typo fix

Changed ligurian to old_ligurian for EU IV compatibility

DrDudelsack committed:
Create test

DrDudelsack committed:
Localization

Localization for the new cultures, bookmark, and partially for the new religions

Idhrendur commented on ImperatorToCk2/Main.java:
```diff
@@ -525,8 +577,10 @@ else if (aq2 < 380) {
+                            int subjectOrNot = Processing.checkSubjectList(tempNum2b,impSubjectInfo);
 
-                            if (impTagInfo.get(tempNum2b)[20] != "none") { //governors
+                            if (impTagInfo.get(tempNum2b)[20] != "none" && subjectOrNot == 9999) { //governors without 9999 check, creates hole
```
From Codacy: Use equals() to compare strings instead of '==' or '!='

Idhrendur commented on ImperatorToCk2/Main.java:
```diff
@@ -419,14 +467,16 @@ else if (aq2 < 380) {
+                            
                             //governor conversion
-                            if (impTagInfo.get(aq4)[20] != "none") {
+                            if (impTagInfo.get(aq4)[20] != "none" && subjectOrNot == 9999) {
```
From Codacy: Use equals() to compare strings instead of '==' or '!='

DrDudelsack committed:
Purged charTest.txt

Purged all references to an early debug test which was causing problems, and laid out the initial groundwork for monument conversion

DrDudelsack committed:
Culture updates

Added mapping for the greco_illyrian culture

DrDudelsack committed:
Initial groundwork for monuments

Added initial groundwork for IR monument conversion, but is currently unfinished

DrDudelsack committed:
Monument mappings

Mappings for I:R monuments and CK II counterparts, currently unused

DrDudelsack committed:
Default religions

Slightly modified the Norse Pagan religion to allow intermarriage with Druidic

DrDudelsack committed:
Localization changes by Thinking_waffle

Thinking_waffle has given almost every new religion localization for their gods and scriptures, as well as changes to the gfx files they use and intermarriage improvements

DrDudelsack committed:
Localization changes by Thinking_waffle

Thinking_waffle has given almost every new religion localization for their gods and scriptures

DrDudelsack committed:
.jar file of compiled .java files to run the converter

Until automization is done, all .jar files have been manually compiled into the .exe.  converter.jar, defaultOutput, and all of the .txt documents not in defaultOutput need to be in the same folder to work

Idhrendur commented on ImperatorToCk2/Importer.java:
```diff
@@ -22,22 +22,25 @@
         FileInputStream fileIn= new FileInputStream(name);
         Scanner scnr= new Scanner(fileIn);
+        
+        String tab = "	";
```
Instead of this, just use \t in the relevant strings.

Idhrendur commented on ImperatorToCk2/Importer.java:
```diff
@@ -280,6 +292,86 @@ else if (qaaa.split("=")[0].equals( "				budget_dates" ) ) {
+
+        int flag = 0;
+        
+        String tab = "	";
```
Ditto

Idhrendur commented on ImperatorToCk2/Importer.java:
```diff
@@ -280,6 +292,86 @@ else if (qaaa.split("=")[0].equals( "				budget_dates" ) ) {
+        output[2] = "9999"; //default for no subject relation
+
+        try {
+            while (endOrNot = true){
```
This is an extra bit of complexity that you can remove.

Idhrendur commented on ImperatorToCk2/Main.java:
```diff
@@ -216,6 +233,13 @@ public static void main (String[] args) throws IOException
                 }else {
                     ck2RegionTotals[ckProvNum] = ck2RegionTotals[ckProvNum] + "~" + impProvRegions[aqq] + "," + impProvInfo[3];
                 }
+                //monuments
```
Go ahead and delete this if you're not using it.

Idhrendur commented on ImperatorToCk2/Processing.java:
```diff
@@ -382,7 +383,7 @@ else if (provName.equals("kyzyl_su")) {
                 output[aqq]=importBaronyName(ck2Dir,aqq,ck2Dir);
                 if (output[aqq].equals("debug")) {
-                    System.out.println(aqq+"is for"+output[aqq]);
+                    //System.out.println(aqq+"is for"+output[aqq]);
```
Just delete the line entirely.
